runtime/common/src/routing.rs

```diff
@@ -0,0 +1,59 @@
+use axelar_gateway_precompile::AxelarKind;
+use cfg_traits::liquidity_pools::{MessageReceiver, MessageSender};
+use cfg_types::domain_address::{Domain, DomainAddress};
+use frame_support::{
+	dispatch::{DispatchResult, DispatchResultWithPostInfo},
+	BoundedVec,
+};
+use pallet_axelar_evm_router::AxelarEvmKind;
+use sp_std::marker::PhantomData;
+
+/// Identification of the router where the message is sent and received
+pub enum RouterId {
+	/// The message must be sent/received by EVM using axelar
+	Axelar(AxelarKind),
+}
```

Comment:
We can not base our decision in the `Domain`, because we can have a router that connect to the same `Domain::Evm` by Axelar or by another kind of middleware. We can not use the `Domain` to distinguish the routers

Reply:
Would it make more sense to have it in `libs` and then use this type in the gateway? Or do we want to specify the `RouterId` as an associated type in the gateway pallet config etc?

Reply:
From the gateway perspective, this type is unknown. It is a runtime configuration type.

If the pallets use directly this type, then the pallet can only be used in runtimes with that type, which would mean runtimes with the same configured router. Also, each pallet would know about other routers ids, and some weird things.

Yeah, as you said, it should be passed as a config type in the gateway

Reply:
Cool, thank you very much!